src/main/webapp/js/script.js

```diff
@@ -0,0 +1,156 @@
+function authenticate() {
+    return gapi.auth2.getAuthInstance()
+        .signIn({scope: "https://www.googleapis.com/auth/youtube.force-ssl"})
+        .then(function() { console.log("Sign-in successful"); },
+              function(err) { console.error("Error signing in", err); });
+}
+
+function loadClient() {
+    gapi.client.setApiKey(config.api_key);
+    return gapi.client.load("https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest")
+        .then(function() { console.log("GAPI client loaded for API"); },
+              function(err) { console.error("Error loading GAPI client for API", err); });
+}
+
+function epoch(timestamp) {
+    const parts = timestamp.split(':');
+    var hours = parseInt(parts[0]);
+    var minutes = parseInt(parts[1]);
+    var seconds = parseInt(parts[2]);
+    var epochTime = hours*3600 + minutes*60 + seconds;
+    return epochTime;
+}
+
+function parseCaptionsIntoJson(response){
+    return new Promise((success, failure) => {
+        var json = {
+            url: 'dummy',
```

Comment:
Is it possible to just grab the url from the front end and placing it here instead of the dummy string?